src/shared/utilities/collectionUtils.ts

```diff
@@ -160,3 +160,40 @@ export async function take<T>(sequence: AsyncIterable<T>, count: number): Promis
 
     return result
 }
+
+export interface IteratorForAWSCallParams<TRequest, TResponse> {
```

Comment:
The "x for" phrasing is unusual. This would be like calling `collectionUtils.ts` , `utilsForCollections.ts` .

Reply:
I think I missed a rename. Should be `getPaginatedAWSCallIterParams` or something similar.